code/rewriter.go

```diff
@@ -130,9 +130,10 @@ func (r *Rewriter) rewriteExpr(expr ast.Expr) error {
 		*ast.FuncType,
 		*ast.InterfaceType,
 		*ast.MapType,
-		*ast.ChanType,
-		*ast.SelectorExpr:
+		*ast.ChanType:
 	// expressions that can not inject failpoint
+	case *ast.SelectorExpr:
```

Comment:
do we have a test case for `ast.SelectorExpr`?

Reply:
Actually
```go
func unittest() {
	type X struct {
		Y int
	}
	// FROM HERE
	func() *X {
		failpoint.Inject("failpoint-name", func(val failpoint.Value) *X {
			return &X{Y: val.(int)}
		})
		return &X{Y: 100}
	}().Y++
	// ^ TO HERE
}
```
is for test case for `ast.SelectorExpr`